app/connector/aws-ddb/src/main/java/io/syndesis/connector/aws/ddb/customizer/DDBConnectorCustomizer.java

```diff
@@ -0,0 +1,139 @@
+
+    //Store options to customize the connector
+    private Map<String, Object> options;
+
+    private static final Logger LOG = LoggerFactory.getLogger(DDBConnectorCustomizer.class);
+
+    @Override
+    public void customize(ComponentProxyComponent component, Map<String, Object> options) {
+
+        this.options = options;
+        component.setBeforeProducer(this::doBeforeProducer);
+
+        component.setAfterProducer(this::doAfterProducer);
+    }
+
+    /**
+     * Extract results and place them on the body.
+     *
+     * @param exchange
+     */
+    protected void doAfterProducer(Exchange exchange) {
+
+        final Message in = exchange.getIn();
+        final Message out = exchange.getOut();
+        final Object item = in.getHeader(DdbConstants.ATTRIBUTES);
+
+        if (item != null) {
+            out.setBody(item);
+        } else if (in.getHeader(DdbConstants.ITEM) != null) {
+            out.setBody(in.getHeader(DdbConstants.ITEM));
+        } else {
+            out.setBody(in.getHeader(DdbConstants.ITEMS));
+        }
+    }
+
+
+    /**
+     * Setup the common headers for all operations.
+     *
+     * @param exchange
+     */
+    protected void doBeforeProducer(Exchange exchange) {
+        exchange.getIn().setHeader(DdbConstants.CONSISTENT_READ, "true");
+        exchange.getIn().setHeader(DdbConstants.RETURN_VALUES, "ALL_OLD");
+
+        customize(exchange, this.options);
+    }
+
+    /**
+     * Customizations for each operation.
+     *
+     * @param exchange
+     * @param options
+     */
+    abstract void customize(Exchange exchange, Map<String,
+            Object> options);
+
+
+    /**
+     * Extract a map from a JSON in a header as AttributeValue. Useful for the headers.
+     *
+     * @param parameterName
+     * @param options
+     * @return
+     */
+    protected Map<String, AttributeValue> getAttributeValueMap(
+            final String parameterName, Map<String, Object> options) {
+        final Map<String, AttributeValue> attributeMap = new HashMap<String, AttributeValue>();
+
+        if (options.containsKey(parameterName)) {
+
+            String element = options.get(parameterName).toString();
+
+            if (!element.isEmpty()) {
+                try {
+                    ObjectMapper mapper = new ObjectMapper();
+
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> attributes = mapper.readValue(element, Map.class);
+
+                    if (attributes.isEmpty() && LOG.isWarnEnabled()) {
+                        LOG.warn("The parameter " + parameterName + " is an empty map.");
+                    }
+
+                    for (Map.Entry<String, Object> attribute : attributes.entrySet()) {
+                        attributeMap.put(attribute.getKey(),
+                                new AttributeValue(attribute.getValue().toString()));
+                        if(LOG.isTraceEnabled()) {
```

Comment:
Personally I don't like log checking, I think it pollutes the readability without any real gain